index.js

```diff
@@ -153,6 +153,9 @@ class ServerlessWebpack {
         .then(this.prepareOfflineInvoke)
         .then(this.wpwatch),
 
+      'before:step-functions-offline:start': () => BbPromise.bind(this)
+        .then(this.validate)
+        .then(this.compile)
```

Comment:
Watch will automatically watch for source changes. Both functionalities (offline watch and invoke local watch) set the base directory for the execution properly, e.g. for serverless-offline the location parameter is set correctly.

If there is something similar for the step-functions that needs to be done, we had to add a different "watch" function for step (e.g. stepWatch).

Reply:
Instead of  using .then(this.validate).then(this.compile) need to create separate function for plugin ?

Reply:
Only _if_ we want to support watch mode and _if_ step-offline needs to know the compiled directory, then you can do:
```
.then(this.validate)
.then(this.prepareStepOffline)   <<<--- see here
.then(this.wpwatch)
```

The new `prepareStepOffline` function should then do something similar as https://github.com/serverless-heaven/serverless-webpack/blob/master/lib/prepareOfflineInvoke.js, only that it prepares the root folder for the step-offline plugin.

Reply:
Need only compile code once, run step-offline and stop process.
I don't need to have watch mode
About compiled directory, user will indicate the path to directory by himself.

Referring to my requirements - need to change something or this code makes sense?

Reply:
Makes sense with the compile, yes.

However:

> About compiled directory, user will indicate the path to directory by himself.

That will not really work out, if you set package: individually. Then the compiled code will then be distributed. I think you need a "prepare" function because for the offline, it has to be set to service packaging (then the output directory is ".webpack/service") like in the prepareOffline.

With individual packaging it is ".webpack/<function-name>" which is not predictable.

Reply:
Sorry, I mean compiled files.
In my plugin user specifies path to function directly
```yaml
custom:
  stepFunctionsOffline:
    FirstLambda: firstLambda/index.handler
```

in case with compiled files it will look like

```yaml
custom:
  stepFunctionsOffline:
    FirstLambda: firstLambda/index.handler
    SecondLambda: .webpack/src/index.handler // in case  individually: true
    SecondLambda: .webpack/service/src/index.handler
```
step-offline does not work on another.

Reply:
`SecondLambda: .webpack/src/index.handler // in case  individually: true`

That is wrong, because the function name is contained in the path. The service packaging path is ok, so it would be better if the webpack plugin would automatically enable service packaging for offline mode - exactly as it is done with serverless-offline.
Then it will work correctly all the time.

Additionally, step-offline could have a location parameter (same as serverless-offline). Then it would be cool and the user does not have to do any own configuration.

Reply:
I mean .webpack/{functionName}/src/index.handler

with your approach I need to rewrite my plugin quite a lot

Reply:
Can you give me the link to your plugin. Maybe I have some time over the weekend to have a look, if we can combine this in an easy and compatible way there.

Reply:
@HyperBrain but looks like your approach very good.
Makes sense to rewrite it.
Only one question, serverless-webpack compiles all functions which described in section *functions* ?

Reply:
Yes

Reply:
Great, 
I update  this pull request to *Is this ready for review?: NO* at this moment.
When I will rewrite code, I will notify you

Reply:
Good idea. I'm already excited 😄